### PR TITLE
Adds OS X support.

### PR DESCRIPTION
**Pull request by cconvey**

### CODE REVIEW

File: ngraph.mk [hunk 1]
```diff
@@ -148,12 +167,11 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
   NGRAPH_COMMON_LIBRARY_LDFLAGS_ := \
     -ltbb \
     -liomp5 \
-    -lmklml_intel \
+    -lmklml \
```
Comment (cconvey) [inline ngraph.mk]: Note to self: Either (a) make this library choice OS-specific, or (b) make it possible to use `libmkml_intel` on OS X.

File: ngraph.mk [hunk 2]
```diff
@@ -148,12 +167,11 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
...
     -lmkldnn
 
   NGRAPH_LDFLAGS_ := \
     -L$(MXNET_LIB_DIR) \
-    -Wl,-rpath-link=$(MXNET_LIB_DIR) \
```
Comment (cconvey) [inline ngraph.mk]: Note to self: make this line-omission OS-specific.

File: ngraph.mk [hunk 3]
```diff
@@ -162,22 +180,19 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
...
     $(NGRAPH_LDFLAGS_) \
-    -Wl,-rpath='$${ORIGIN}/../lib' \
-    -Wl,--as-needed
+    -Wl,-rpath,@loader_path/../lib
 
   NGRAPH_LDFLAGS_FOR_CPP_UNIT_TESTS_PROG := \
     $(NGRAPH_LDFLAGS_) \
-    -Wl,-rpath='$${ORIGIN}/../../../lib' \
-    -Wl,--as-needed
+    -Wl,-rpath,@loader_path/../../../lib
```
Comment (cconvey) [inline ngraph.mk]: Not to self: Make all oft the changes above be OS-specific.

**Comment by cconvey** [comment]
Save for future PRs?

- Make sure we're using the intended math libraries at runtime.  (I.e., re-audit them since OS X's `dyld` != Linux's `ldd`).

- Add support for `libmklml_intel` on OS X.

File: cmake/ngraph.cmake [hunk 4]
```diff
@@ -36,10 +35,10 @@ endif(USE_NGRAPH_DISTRIBUTED)
 ExternalProject_Add(
 	ext_ngraph
 	GIT_REPOSITORY https://github.com/NervanaSystems/ngraph.git
-	GIT_TAG v0.6.0
+	GIT_TAG "v0.6.0"
```
Comment (mbrookhart) [inline cmake/ngraph.cmake]: Cmake on Mac peculiarity?

File: ngraph.mk [hunk 5]
```diff
@@ -145,40 +165,72 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
   # nGraph provides some libraries that may compete with other libraries already installed
   # on the system. This provides the link-flags that, if provided early enough on the static-linker
   # command line, should ensure that the nGraph-supplied version is preferred.
+  ifeq ("$(shell uname)","Darwin")
+    NGRAPH_MKLML_LIB_ := mklml
```
Comment (mbrookhart) [inline ngraph.mk]: Does MKLDNN fall back to libmklml on Mac?

File: ngraph.mk [hunk 6]
```diff
@@ -145,40 +165,72 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
...
+
+    ifeq ($(USE_NGRAPH_DISTRIBUTED), 1)
+      $(error "USE_NGRAPH_DISTRIBUTED=1 is not yet supported on OS X.")
+    endif
+
+    # The static-link-time flags for creating .so/.dylib files that will dynamically link to nGraph's libs
+    # at runtime.
+    NGRAPH_LDFLAGS_FOR_SHARED_LIBS := \
+      $(NGRAPH_LDFLAGS_) \
+      -Wl,-rpath,@loader_path
```
Comment (mbrookhart) [inline ngraph.mk]: Why no --as-needed?

File: cmake/ngraph.cmake (hunk 4 above)
Comment (cconvey) [reply cmake/ngraph.cmake]: No, a leftover from a change I'd experimented with, coupled with my tendency for excessive caution in languages like Bash and CMake.  Can remote if you want.

File: ngraph.mk (hunk 5 above)
Comment (cconvey) [reply ngraph.mk]: Short answer: I'm not sure.

tl;dr:  I was planning to tackle this question in a follow-on PR, but I didn't want to hold things up.  IIRC this change was prompted by a combination of failed builds and of seeing the logic [in this file](https://github.com/NervanaSystems/ngraph-mxnet/blob/master/cmake/MklDnn.cmake).

File: ngraph.mk (hunk 6 above)
Comment (cconvey) [reply ngraph.mk]: Short answer: It's not provided in Apple's version of `ld`.

Longer answer: I'm planning to audit the runtime binding of math libraries in a future PR.  But given how messy it got on Linux, I didn't want to let that hold up this current PR.

File: ngraph.mk (hunk 5 above)
Comment (cconvey) [reply ngraph.mk]: @mbrookhart : A partial answer to your question: [https://mxnet.apache.org/install/build_from_source.html#math-library-selection](https://mxnet.apache.org/install/build_from_source.html#math-library-selection).
